plugin.js

```diff
@@ -0,0 +1,139 @@
+const path = require("path");
+const fs = require("fs");
+const shell = require("shelljs");
+const { JSDOM } = require("jsdom");
+
+const Config = require("./Config.js");
+const { PluginScope } = require("./Enums.js");
+
+function setCurr(component, targetPath, subTaskFlag=false) {
+	let obj = {...component}, isCurrentItem = false;
+
+	if(obj.unit_type === "aim")
+	{
+		subTaskFlag = true;
+		isCurrentItem = true;
+	}
+
+	if(!subTaskFlag)
+	{
+		obj = component.menuItemInfo(targetPath);
+	}
+
+	if(obj.unit_type === "lu")
+	{
+		obj.units = [...obj.units.map((subComponent) => setCurr(subComponent, targetPath, true))];
+	}
+
+	return { ...obj, isCurrentItem: isCurrentItem };
+};
+
+class Plugin {
+
+  static getConfigFileName(options_env) {
+    const env = options_env || BuildEnvs.TESTING;
+    const pluginConfigFile = `./plugin-config.${env}.js`;
+    return pluginConfigFile;
+  }
+
+  static processExpScopePlugins(exp_info, hb, lab_data, options) {
+    const env = options.env || BuildEnvs.TESTING;
+    const pluginConfigFile = `./plugin-config.${env}.js`;
+    const pluginConfig = require(pluginConfigFile);
+
+    const expScopePlugins = pluginConfig.filter(
+      (p) => p.scope === PluginScope.EXPERIMENT
+    );
+
+    let plugins = [];
+    shell.exec('mkdir plugins');
+    expScopePlugins.forEach((plugin) => {
+	    try {
+		    if(!fs.existsSync(path.join('plugins', plugin.id)))
+		    {
+			    shell.cd('plugins');
+			    shell.exec('git clone ' + plugin.repo);
+			    shell.exec(`rm -rf ${path.join(plugin.id, '.git')}`);
+			    shell.cd('..');
+		    }
```

Comment:
In the past few lines, the treatments of the plugins directory and the plugin.id directory are contrasting.
We are assuming that the plugins directory doesn't exist but we still check for the plugin.id directory's existence. If the parent directory doesn't exist, the child directory obviously wouldn't unless you expect some plugin config to be duplicate.
I'd suggest the following:
1. Check for the existence of the plugins directory before creation. Create it if it doesn't exist.
2. If the plugin.id directory does not exist, create it and shallow clone (--depth=1) it.
3. If the plugin.id does exist, pull it so it gets updated.
4. In the clean command on package.json, ensure that the plugins directory is deleted.